src/extension/kernel.ts

```diff
@@ -691,6 +692,8 @@ export class Kernel implements Disposable {
     }
 
     TelemetryReporter.sendTelemetryEvent('cell.startExecute')
+    // todo(sebastian): rewrite to use non-blocking impl
+    await getEventReporter().reportExecution(cell)
```

Comment:
@sourishkrout The await here came from your patch. Did you mean to include an await and make it blocking? If we call await here then I assume we block until the event is full reported? Whereas I thought if we don't use await here then reportExecution would perform the RPC asynchronously and not block actual execution?

Reply:
Thanks for the explanation!

Reply:
@sourishkrout Is this good to merge?

Reply:
Yep, approving the PR now.